File: scripts/qtools/qstr.rb
module QStr
  # Generates a 6-character unique identifier (suuid) consisting of random
  # upper and lower case letters and numbers.
  # Example: "nO57aL"
  def self.generate_suuid
    characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
    suuid = ""
    6.times do
      suuid << characters[rand(characters.length)]
    end
    suuid
  end
end

File: scripts/qtools/qstr.php
<?php

/**
 * String utility methods
 */
class qstr {
    /**
     * Generates a 6-character unique identifier (suuid) consisting of random
     * upper and lower case letters and numbers.
     * Example: "nO57aL"
     * 
     * @return string The generated suuid
     */
    public static function generateSuuid(): string {
        $characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
        $suuid = "";
        for ($i = 0; $i < 6; $i++) {
            $randomIndex = rand(0, strlen($characters) - 1);
            $suuid .= $characters[$randomIndex];
        }
        return $suuid;
    }
} 